mongo/dataset_store.go

```diff
@@ -27,6 +27,11 @@ func (m *Mongo) GetDatasetsByBasedOn(ctx context.Context, id string, offset, lim
 		},
 	}
 
+	// added for total_count to return true value.
+	if authorised {
+		filter["current"] = bson.M{"$exists": true}
+	}
+
```

Comment:
Isn't this the other way around?

Non-authorised queries, require that the current edition must exist?

Reply:
Yes, you are correct, changed to reflect unauthorized